packages/amplify-provider-awscloudformation/src/__tests__/push-resources.test.ts

```diff
@@ -0,0 +1,78 @@
+import { pathManager } from 'amplify-cli-core';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import { getCfnFiles } from '../push-resources';
+import { mocked } from 'ts-jest/utils';
+import glob from 'glob';
+
+// Mock data
+const dummyCFNFiles: readonly string[] = ['fileA', 'fileB'];
+const dummyNestedCFNFiles: readonly string[] = ['nested_fileA', 'nested_fileB'];
+const testBackendDirPath = path.join('backendTest', 'path');
+
+// Mocks
+jest.mock('amplify-cli-core');
+jest.mock('fs-extra');
+jest.mock('../system-config-manager');
+jest.mock('glob');
+
+const pathManager_mock = mocked(pathManager);
+pathManager_mock.getBackendDirPath.mockReturnValue(testBackendDirPath);
+
+const glob_mock = glob as jest.Mocked<typeof glob>;
+glob_mock.sync.mockImplementation((pattern) => {
+  switch (pattern) {
+    case '*template*.+(yaml|yml|json)':
+      return [...dummyCFNFiles];
+    case 'stacks/*.+(yaml|yml|json)':
+      return [...dummyNestedCFNFiles];
+  }
+  return [];
+});
+
+const fs_mock = mocked(fs);
+fs_mock.lstatSync.mockImplementation((_path, _options) => {
+  return {
+    isDirectory: jest.fn().mockReturnValue(true),
+  } as unknown as fs.Stats;
+});
+const fsSpy = jest.spyOn(fs_mock, 'existsSync');
+
+describe('test getCfnFiles', () => {
+  it.only('does return API category stack from the build directory', async () => {
+    // The build path generated have files, so should be returned.
+    fs_mock.existsSync.mockReturnValue(true);
+    glob_mock.sync.mockImplementationOnce(() => [...dummyCFNFiles]);
+
+    // call the function under test
+    const { resourceDir, cfnFiles } = getCfnFiles('api', 'api-resource-name');
+    expect(resourceDir).toStrictEqual('backendTest/path/api/api-resource-name/build');
+    expect(fsSpy).toHaveBeenCalledWith('backendTest/path/api/api-resource-name/build');
+    expect(cfnFiles).toEqual(dummyCFNFiles);
+  });
+
+  it.only('does return API category stack from the build directory along with all nested stacks', async () => {
```

Comment:
You have two tests with `only` and one without. Is this intentional? The test on line 67 will not run.

Reply:
it's not, good catch. remnant of my testing. Updated